code/game/machinery/computer/medical.dm

```diff
@@ -315,20 +331,35 @@
 							if ((!( t1 ) || !( src.authenticated ) || usr.stat || usr.restrained() || (!in_range(src, usr) && (!istype(usr, /mob/living/silicon))) || src.active1 != a1))
 								return
 							src.active1.fields["dna"] = t1
+					/*
 					if("vir_name")
 						var/datum/data/record/v = locate(href_list["edit_vir"])
 						if (v)
 							var/t1 = copytext(sanitize(input("Please input pathogen name:", "VirusDB", v.fields["name"], null)  as text),1,MAX_MESSAGE_LEN)
 							if ((!( t1 ) || !( src.authenticated ) || usr.stat || usr.restrained() || (!in_range(src, usr) && (!istype(usr, /mob/living/silicon))) || src.active1 != a1))
 								return
 							v.fields["name"] = t1
+					*/
+					if("vir_nickname")
+						var/datum/data/record/v = locate(href_list["edit_vir"])
+						if (v)
+							var/t1 = copytext(sanitize(input("Please input pathogen nickname:", "VirusDB", v.fields["nickname"], null)  as text),1,MAX_MESSAGE_LEN)
+							if ((!( t1 ) || !( src.authenticated ) || usr.stat || usr.restrained() || (!in_range(src, usr) && (!istype(usr, /mob/living/silicon))) || src.active1 != a1))
+								return
+							v.fields["nickname"] = t1
+							temp = pathogen_dat(v)
 					if("vir_desc")
 						var/datum/data/record/v = locate(href_list["edit_vir"])
 						if (v)
 							var/t1 = copytext(sanitize(input("Please input information about pathogen:", "VirusDB", v.fields["description"], null)  as message),1,MAX_MESSAGE_LEN)
 							if ((!( t1 ) || !( src.authenticated ) || usr.stat || usr.restrained() || (!in_range(src, usr) && (!istype(usr, /mob/living/silicon))) || src.active1 != a1))
 								return
 							v.fields["description"] = t1
+							temp = pathogen_dat(v)
+					if("danger_vir")
+						var/datum/data/record/v = locate(href_list["edit_vir"])
+						if (v)
+							src.temp = text("<B>Pathogen Dangerousness:</B><BR>\n\t<A href='?src=\ref[];temp=1;danger_vir=dangerous;edit_vir=\ref[v]'>*DANGEROUS*</A><BR>\n\t<A href='?src=\ref[];temp=1;danger_vir=undetermined;edit_vir=\ref[v]'>Undetermined</A><BR>\n\t<A href='?src=\ref[];temp=1;danger_vir=safe;edit_vir=\ref[v]'>Safe</A><BR>", src, src, src)
```

Comment:
this file in general is overflowing with srcs

Reply:
what's the point of using the text() proc here anyway?

Reply:
it used to be in vogue because you put the vars at the end, personally I think it's illegible compared to just including them in brackets - another good thing to eliminate

Reply:
well there's like 50 of them in this file so I'll just remove the src. for now